fix(cache): remove unused webserver config & handle trailing slashes

Usiel opened the pull request:
### SUMMARY

Values for `WEBDRIVER_BASEURL` must end with a slash for the cache warmup to work. Our mdx files sometimes show example values without a trailing slash, which is quite confusing. We can adapt the `fetch_url` task to handle either case (trailing or non-trailing slash) by using our existing util function `get_url_path(...)`.

Related to this (`WEBDRIVER_BASEURL`): We stopped using `SUPERSET_WEBSERVER_PROTOCOL`, `SUPERSET_WEBSERVER_ADDRESS`, `SUPERSET_WEBSERVER_PORT` with https://github.com/apache/superset/pull/21076. I'm removing the keys from the base config as there does not seem to be any immediate need for them: We already have `WEBDRIVER_BASEURL` (and the `_USER_FRIENDLY` variant), which are in use all over the project.

### TESTING

Added an integration test to ensure the URL is generated correctly.

Full-text search for the deleted config keys :)

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue: 
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

Usiel committed:
Remove unused SUPERSET_WEBSERVER_ config

We stopped using SUPERSET_WEBSERVER_PROTOCOL, ..._ADDRESS, ..._PORT with https://github.com/apache/superset/pull/21076. Removing the config from the example config as there does not seem to be any immediate need for it, we already have WEBDRIVER_BASEURL (and the _USER_FRIENDLY variant).

Usiel commented:
Just a friendly reminder for this PR (no functional changes).

sfirke commented:
Does WEBDRIVER_BASEURL need to end in a slash?  Mine does not in my config and reports etc. all work correctly,  I have `WEBDRIVER_BASEURL="http://superset_app:8088"`

Usiel commented:
> Does WEBDRIVER_BASEURL need to end in a slash? Mine does not in my config and reports etc. all work correctly, I have `WEBDRIVER_BASEURL="http://superset_app:8088"`

Thanks for checking. You are right, reports are fine, as the function that is called ultimately uses a library that is clever enough to handle both cases (ending in a slash or not), see [urls.py#L32](https://github.com/apache/superset/blob/2.1.0/superset/utils/urls.py#L32).

However, your cache warmup might be broken, because [cache.py#L43](
https://github.com/apache/superset/blob/2.1.0/superset/tasks/cache.py#L43) does not handle the base URL ending with a slash. I'm happy to improve this, but not sure if you all would prefer a separate PR for that.

sfirke commented:
I searched the codebase for the arguments removed by this PR and agree, they are not in use and you got all mentions of them.  I can't approve PRs but would approve this one 😁 

Re: 
> However, your cache warmup might be broken, because [cache.py#L43](https://github.com/apache/superset/blob/2.1.0/superset/tasks/cache.py#L43) does not handle the base URL ending with a slash. I'm happy to improve this, but not sure if you all would prefer a separate PR for that.

Indeed I've never gotten cache warmup to work 😆  I did not dig into the sub-functions but I see what you're saying, that there's processing done by that library that's not present in cache.py.  I think that would be a nice improvement to add it to cache.py#43 to bring it into parity with urls.py, and a separate PR.

Antonio-RiveroMartnez commented on docs/docs/installation/alerts-reports.mdx:
```diff
@@ -163,9 +163,9 @@ WEBDRIVER_OPTION_ARGS = [
-WEBDRIVER_BASEURL = "http://superset:8088"
-# This is the link sent to the recipient. Change to your domain, e.g. https://superset.mydomain.com
-WEBDRIVER_BASEURL_USER_FRIENDLY = "http://localhost:8088"
+WEBDRIVER_BASEURL = "http://superset:8088/"
```
Hey! Thanks for the fix! 🎉  just a comment, could we leave this as is and instead address the correct handling of both cases (with our without trailing /) in the get_url method?

sfirke commented:
Bumping this since I see someone is getting confused by the out-of-date docs that this would have fixed: https://github.com/apache/superset/pull/22849  Maybe merge this to fix the docs, then address the suggestion from @Antonio-RiveroMartnez about fixing trailing URLs in a different PR?

Usiel committed:
Handle WEBDRIVER_BASEURL slashes

As suggested, we handle both cases (trailing slash, no trailing slash) for the WEBDRIVER_BASEURL. We use the existing `superset.utils.urls` `get_url_path(...)` function to manage everything.

Usiel commented:
Rebase on latest `master` - also, another run on the tests, as there were some failures (unrelated imo).

sfirke commented:
Well I just got burned by the trailing URL thing 😩 On the plus side, I'm now a committer so could approve this PR.  @Usiel do you have interest in picking it up?  It would probably need a rebase.

Usiel commented:
Happy to give this another shot, probably sometime end of next week :)

Usiel committed:
Adds test to ensure trailing slashes are handled for fetch_url task

Usiel commented:
I rebased on the latest `master`. The issue still persists even though there was some refactoring. I fixed the `fetch_url` task to use the existing util function `get_url_path`, which handles trailing slashes.

Ready for another review

john-bodley commented:
cc @eschutho and @michael-s-molina as these unused config variables were never previously referenced in `UPDATING.md` as being deprecated and/or obsolete.

Usiel commented:
Friendly ping :) Is there still interest in this? I can rebase on the latest if yes.

sfirke commented:
Yes please @Usiel !  If you rebase I will do my best to ping the right committer to review and merge.

Usiel commented:
> Yes please @Usiel ! If you rebase I will do my best to ping the right committer to review and merge.

Rebased (no conflicts), thanks!

sfirke commented:
@john-bodley or @michael-s-molina could you take a look at this, maybe during the daily PR review?  It is ready for merge to my eyes but I don't know enough to evaluate the tests.

I don't think it's a problem that in the past, there was a failure to document the removal of the now-unused variables that this cleans up.  But if you like, I can make a follow-up PR that adds it retroactively to the 2.0.0 release notes as I propose in my comment above.

michael-s-molina reviewed (approved):
LGTM
